src/databricks/labs/remorph/reconcile/utils.py

```diff
@@ -0,0 +1,73 @@
+from pyspark.sql import DataFrame
+from pyspark.sql.functions import col, lit, expr
+
+from databricks.labs.remorph.reconcile.recon_config import ReconcileOutput
+
+
+def reconcile_data(source: DataFrame, target: DataFrame, join_cols: list[str],
+                   report_type: str) -> ReconcileOutput:
+    join = join_cols if report_type in {"data", "all"} else "hash_value_recon"
```

Comment:
It isn't clear how the join expressions know about the `s` and `t` aliases, won't it break id someone decides to change them to, say, `source` and `target` without changing the code that calls `reconcile_data`?
Moreover we may at some point need to dynamically set these aliases (eg. for avoiding name clashes).
I suggest we make `join_cols` a function that takes two strings (the alias names) and returns a list of strings (expressions) and also use string interpolation whenever instead of hard-coding the alias names in the rest of `reconcile_data` (eg. `f"{targetAlias}.*"` instead of `"t.*"`.

Reply:
I have updated the string interpolation.But didn't get the context on join column alias.In our case we will have always the join cols as same name in src and tgt. So we no need to alias the join cols.We can use list[join_cols] and this works well.